docs/custom-components.mdx

```diff
@@ -0,0 +1,286 @@
+---
+title: Customize the template
+---
+<!--docsignore
+import { Callout } from 'fumadocs-ui/components/callout';
+import { DocsPage } from 'fumadocs-ui/page';
+docsignore-->
+
+Use the info in this guide to customize the template to create your own custom service. Check out the [WAVS design considerations](/design) page to learn which use-cases WAVS is best suited for.
+
+## Trigger
+
+The simple onchain trigger contract in the template is used to emit triggers to the WAVS service. The trigger data is encoded as a bytes object. When a WAVS component receives this trigger, it uses the `decode_event_log_data!` macro from the [`wavs-wasi-chain`](https://docs.rs/wavs-wasi-chain/latest/wavs_wasi_chain/all.html#functions) crate to decode the event data for processing.
+
+Each trigger has a unique ID that links it to its data. The component processes the trigger data and returns results. Results are submitted back to the blockchain via the submission contract.
```

Comment:
heads up - this all makes sense for the example, and wavs itself does not have a concept of a "trigger id", but in 0.4 there _is_ going to be a new concept of "event id" which will be independent of the "trigger id" here. just a heads up so you're not surprised when we'll need to disambiguate them :)